### PR TITLE
v1.0.0

### PR DESCRIPTION
**Pull request by mamoonraja**
This is the first major release for Ruby SDK 💎, this PR includes:

- Revamped Auth methods to use AUthenticators.
- Latest version of the core.
- Updates to gems
- Updated examples
- Regeneration of services based on current API definitions
- Support for Visual Recognition V4.

## To Do's
- [x] Migration Guide
- [x] Make sure bumpversion and versions in the main file are correct so that semantic release will work just fine.
- [x] Update core gem to `v1.0.0.`

Closes https://github.com/watson-developer-cloud/ruby-sdk/issues/56

### CODE REVIEW

File: .bumpversion.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 0.20.1
+current_version = 1.0.0.rc1
```
[the remark this answers is not in the document]
Comment (mamoonraja) [reply .bumpversion.cfg]: yeah planning to change bump version and other places which might effect semantic release before merging, will update with the changes for new version of core gem.

**Comment by mamoonraja** [comment]
@mediumTaj Regenerated and looks like it is up to date, making changes for core version and `bumpversion` now

**Comment by mediumTaj** [comment]
Mamoon, can you regenerate discovery using API def `ef9010ab7de1b1fc7628a340cab118e2f00dd51d`? There were some issues with some of the cpd only functionality that was added

**Comment by mediumTaj** [comment]
We also need `MIGRATION-V1.md` here

**Comment by mamoonraja** [comment]
@mediumTaj Working on it, should be in soon.

**Comment by mamoonraja** [comment]
@SirSpidey wrapping it up, it's in to-do list above :)

**Comment by mamoonraja** [comment]
@SirSpidey @mediumTaj Migration guide is in.

File: lib/ibm_watson/discovery_v1.rb [hunk 2]
```diff
@@ -1833,110 +1729,43 @@ def federated_query_notices(environment_id:, collection_ids:, filter: nil, query
     end
 
     ##
-    # @!method query_entities(environment_id:, collection_id:, feature: nil, entity: nil, context: nil, count: nil, evidence_count: nil)
-    # Knowledge Graph entity query.
-    # See the [Knowledge Graph
-    #   documentation](https://cloud.ibm.com/docs/services/discovery?topic=discovery-kg#kg)
-    #   for more details.
+    # @!method get_autocompletion(environment_id:, collection_id:, field: nil, prefix: nil, count: nil)
```
Comment (mediumTaj) [inline lib/ibm_watson/discovery_v1.rb]: prefix here should be required. did you regenerate with `ef9010ab7de1b1fc7628a340cab118e2f00dd51d`

Comment (mediumTaj) [reply lib/ibm_watson/discovery_v1.rb]: disregard this comment - i made it before i asked you about regenerating

**Pull request merged**